src/App.js

```diff
@@ -1,23 +1,148 @@
-import React from 'react';
-import SentenceContainer from './components/Sentence/SentenceContainer.js'
-import SettingsContainer from './components/SettingsContainer.js'
-import logo from './logo.svg';
+import React, { Component } from 'react';
+import Sentence from './components/Sentence/Sentence'
+import Settings from './components/Settings'
 import './App.css';
+import {allScales, getAccidentalFromNoteString, isInArray, keyFormulas, nashvilleNumbers} from "./Constants";
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-          <SentenceContainer/>
-      </header>
-        <div>
-            TODO:
-            <li>- SHow guess value in setence with green or red correct/incorrect</li>
-                <li>- settings/config</li>
-        </div>
-        <SettingsContainer/>
-    </div>
-  );
+class App extends Component
+{
+
+    getAllKeysWithTrueValue = (obj) => {
+        let out = [];
+        let keys = Object.keys(obj);
+        for (let i of keys) {
+            if (obj[i] === true){
+                out.push(i);
+            }
+        }
+        return out;
+    }
+
+    componentWillMount() {
```

Comment:
@groman00 makes sense. It may be more logical in the constructor (which I don't see in the component).

`componentWillMount` will be called every time the component gets added to the DOM tree. In this case, it will only be once (theoretically) since you have one instance of `App`. To be more specific, it is called right before the component gets rendered

Reply:
@groman00 Calling componentWillMount() in the constructor?  I _think_ I tried that but it was a while ago now... let me try it again.

Reply:
@tmbridge No, call `resetCurrentKeysByFilters` in your constructor.  Never call a lifecycle method directly.